src/Interpreters/Session.cpp

```diff
@@ -164,7 +164,6 @@ class NamedSessionsStorage
     void releaseAndCloseSession(const UUID & user_id, const String & session_id, std::shared_ptr<NamedSessionData> & session_data)
     {
         std::unique_lock lock(mutex);
-        scheduleCloseSession(*session_data, lock);
```

Comment:
Perhaps we could also remove the session from the `close_time_buckets` map, but maybe that's not necessary.